modules/apps/foundation/users-admin/users-admin-web/src/main/resources/META-INF/resources/view_flat_users.jsp

```diff
@@ -266,7 +266,13 @@ boolean showRestoreButton = (searchTerms.getStatus() != WorkflowConstants.STATUS
 		}
 		else {
 			if (filterManageableOrganizations && !UserPermissionUtil.contains(permissionChecker, ResourceConstants.PRIMKEY_DNE, ActionKeys.VIEW)) {
-				userParams.put("usersOrgs", ArrayUtil.toLongArray(user.getOrganizationIds()));
+				long[] organizationIds = user.getOrganizationIds();
+
+				if (ArrayUtil.isEmpty(organizationIds)) {
+					organizationIds = new long[] {0};
```

Comment:
You are right, sorry, I misread it.

However I must say that now it seems even more strange. I mean it's like a magic value (an array with one element which is zero) which creates further questions such as "Does the value of the array have to be 0? Could it be other value?". This is subject to be misunderstood by future developers (including our future selves) who could introduce bugs as a result.

To avoid this, I think we might want to introduce a constant which represents NO_ORGANIZATIONS and pass it over. The same constant should then be used in the finder to represent that value.

What do you think?